contrib/endpoints/src/api_manager/service_control/aggregated.cc

```diff
@@ -497,56 +537,17 @@ void Aggregated::Call(const RequestType& request, ResponseType* response,
     max_report_size_ = request_body.size();
   }
 
-  auto token = (typeid(RequestType) == typeid(AllocateQuotaRequest))
-                   ? sa_token_quota_
-                   : sa_token_;
-
   http_request->set_url(url)
       .set_method("POST")
-      .set_auth_token(GetAuthToken(token))
+      .set_auth_token(GetAuthToken(request))
```

Comment:
I am not sure how I can remove the request variable? 
Without the argument for the function template, there is no way to pass RequestType.  

I have tried and I got following compiler errors.

contrib/endpoints/src/api_manager/service_control/aggregated.cc: In member function 'void google::api_manager::service_control::Aggregated::Call(const RequestType&, ResponseType*, google::service_control_client::TransportDoneFunc, google::api_manager::cloud_trace::CloudTraceSpan*)':
contrib/endpoints/src/api_manager/service_control/aggregated.cc:537:36: error: no matching function for call to 'google::api_manager::service_control::Aggregated::GetAuthToken()'
       .set_auth_token(GetAuthToken())
                                    ^
contrib/endpoints/src/api_manager/service_control/aggregated.cc:537:36: note: candidate is:
In file included from contrib/endpoints/src/api_manager/service_control/aggregated.cc:17:0:
./contrib/endpoints/src/api_manager/service_control/aggregated.h:126:22: note: template<class RequestType> const string& google::api_manager::service_control::Aggregated::GetAuthToken()
   const std::string& GetAuthToken();
                      ^
./contrib/endpoints/src/api_manager/service_control/aggregated.h:126:22: note:   template argument deduction/substitution failed:
contrib/endpoints/src/api_manager/service_control/aggregated.cc:537:36: note:   couldn't deduce template parameter 'RequestType'
       .set_auth_token(GetAuthToken())